### PR TITLE
[MBL-16890][Student] - Test folder control flow in user groups

### PR DESCRIPTION
**Pull request by kdeakinstructure**
Implement user group folder control flow (create folder, upload file, create subfolder, open folder).
Add some UserGroup related classes.
Add some page object methods.
Initial commit (will be continued with the exact test case)

refs: MBL-16890
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/androidTest/java/com/instructure/student/ui/pages/CourseBrowserPage.kt [hunk 1]
```diff
@@ -125,6 +130,10 @@ class CourseBrowserPage : BasePage(R.id.courseBrowserPage) {
         initialBrowserTitle.assertHasText(course.name)
     }
 
+    fun assertTitleCorrect(group: GroupApiModel) {
```
Comment (kristofnemere) [inline apps/student/src/androidTest/java/com/instructure/student/ui/pages/CourseBrowserPage.kt:133]: This method should be in GroupBrowserPage

File: apps/student/src/androidTest/java/com/instructure/student/ui/e2e/usergroups/UserGroupFilesE2ETest.kt [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                "Assert that the '$testFolderName2' folder's size is 0, because we just created it.")
+        fileListPage.assertItemDisplayed(testFolderName2)
+        fileListPage.assertFolderSize(testFolderName2, 0)
+
+        Log.d(STEP_TAG, "Select '$testFolderName2' folder and assert that the empty view is displayed.")
+        fileListPage.selectItem(testFolderName2)
+        fileListPage.assertViewEmpty()
+    }
+
+    private fun setupFileOnDevice(fileName: String): Uri {
```
Comment (kristofnemere) [inline apps/student/src/androidTest/java/com/instructure/student/ui/e2e/usergroups/UserGroupFilesE2ETest.kt:137]: We have this method already somewhere else, I would put that in a util class and use it from all the other places

**Review by kristofnemere — CHANGES_REQUESTED** [review]

File: apps/student/src/androidTest/java/com/instructure/student/ui/pages/CourseBrowserPage.kt (hunk 1 above)
Comment (kdeakinstructure) [reply apps/student/src/androidTest/java/com/instructure/student/ui/pages/CourseBrowserPage.kt:133]: Sure, and so the other Group related methods. Nice catch, thanks.

**Review by tamaskozmer — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kristofnemere — APPROVED** [review]

**Pull request merged**